javagen/src/main/java/com/azure/autorest/template/PatchModelTemplate.java

```diff
@@ -0,0 +1,290 @@
+            // serialization. For now, it's needed to handle the rare scenario where the strong type is a non-Java
+            // base type.
+            imports.add(JacksonAdapter.class.getName());
+        }
+
+        String lastParentName = model.getName();
+        ClientModel parentModel = ClientModelUtil.getClientModel(model.getParentModelName());
+        while (parentModel != null && !lastParentName.equals(parentModel.getName())) {
+            imports.addAll(parentModel.getImports());
+            lastParentName = parentModel.getName();
+            parentModel = ClientModelUtil.getClientModel(parentModel.getParentModelName());
+        }
+
+        List<ClientModelPropertyReference> propertyReferences = this.getClientModelPropertyReferences(model);
+        if (JavaSettings.getInstance().isOverrideSetterFromSuperclass()) {
+            propertyReferences.forEach(p -> p.addImportsTo(imports, false));
+        }
+
+        if (!CoreUtils.isNullOrEmpty(model.getPropertyReferences())) {
+            if (JavaSettings.getInstance().getClientFlattenAnnotationTarget() == JavaSettings.ClientFlattenAnnotationTarget.NONE) {
+                model.getPropertyReferences().forEach(p -> p.addImportsTo(imports, false));
+            }
+            propertyReferences.addAll(model.getPropertyReferences());
+        }
+
+        model.addImportsTo(imports, settings);
+
+        imports.add("com.azure.core.implementation.Option");
+
+        javaFile.declareImport(imports);
+
+        javaFile.javadocComment(settings.getMaximumJavadocCommentWidth(), (comment) ->
+                comment.description(model.getDescription()));
+
+        boolean hasDerivedModels = !model.getDerivedModels().isEmpty();
+//        handlePolymorphism(model, hasDerivedModels, settings.isDiscriminatorPassedToChildDeserialization(), javaFile);
+
+//        if (settings.getClientFlattenAnnotationTarget() == JavaSettings.ClientFlattenAnnotationTarget.TYPE && model.getNeedsFlatten()) {
+//            javaFile.annotation("JsonFlatten");
+//        }
+
+        ArrayList<JavaModifier> classModifiers = new ArrayList<>();
+        if (!hasDerivedModels && !model.getNeedsFlatten()) {
+            if (!settings.isFluent() || !model.getName().endsWith("Identity")) {    // bug https://github.com/Azure/azure-sdk-for-java/issues/8372
+                classModifiers.add(JavaModifier.Final);
+            }
+        }
+
+        String classNameWithBaseType = model.getName();
+        if (model.getParentModelName() != null) {
+            classNameWithBaseType += String.format(" extends %1$s", model.getParentModelName());
+        }
+        final String patchClassNameWithBaseType = classNameWithBaseType + "Patch";
+
+        if (model.getProperties().stream().anyMatch(p -> !p.getIsReadOnly())
+                || propertyReferences.stream().anyMatch(p -> !p.getIsReadOnly())) {
+            javaFile.annotation("Fluent");
+        } else {
+            javaFile.annotation("Immutable");
```

Comment:
I don't think you will ever have a Patch model to be `Immutable` (if all its property is read-only, then what's the point of PATCH?).